streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -233,26 +218,51 @@ InternalTopologyBuilder builder() {
     }
 
     /**
-     * Similar to shutdownTasksAndState, however does not close the task managers, in the hope that
-     * soon the tasks will be assigned again
+     * Closes suspended active tasks and standby tasks that were reassigned at the end of a rebalance.
+     *
+     * @return list of changelog topic partitions from revoked tasks
      * @throws TaskMigratedException if the task producer got fenced (EOS only)
      */
-    void suspendTasksAndState()  {
-        log.debug("Suspending all active tasks {} and standby tasks {}", active.runningTaskIds(), standby.runningTaskIds());
+    List<TopicPartition> closeRevokedTasks() {
+        log.debug("Closing revoked active tasks {} and standby tasks {}", revokedActiveTaskIds(), revokedStandbyTaskIds());
+
+        final List<TopicPartition> revokedChangelogPartitions = new ArrayList<>();
 
+        active.closeNonAssignedSuspendedTasks(assignedActiveTasks, revokedChangelogPartitions);
+        standby.closeReassignedStandbyTasks(revokedStandbyTasks, revokedChangelogPartitions);
+
+        // remove any revoked partitions from the restore consumer's assignment and changelog reader
+        final Set<TopicPartition> updatedAssignment = new HashSet<>(restoreConsumer.assignment());
+        updatedAssignment.removeAll(revokedChangelogPartitions);
+        restoreConsumer.assign(updatedAssignment);
+        changelogReader.remove(revokedChangelogPartitions);
+
+        return revokedChangelogPartitions;
+    }
+
+    /**
+     * Similar to shutdownTasksAndState, however does not close the task managers, in the hope that
+     * soon the tasks will be assigned again.
+     * @throws TaskMigratedException if the task producer got fenced (EOS only)
+     */
+    void suspendActiveTasksAndState(final Collection<TopicPartition> revokedPartitions)  {
```

Comment:
We can let this function to return the list of suspended tasks, so that in rebalance listener we can print it in log4j entry; by doing this we do not need the `suspendedActiveTaskIds`.